components/org.wso2.carbon.identity.api.server.script.library/org.wso2.carbon.identity.api.server.script.library.v1/src/main/java/org/wso2/carbon/identity/api/server/script/library/v1/core/ServerScriptLibrariesService.java

```diff
@@ -0,0 +1,439 @@
+import org.wso2.carbon.identity.api.server.script.library.v1.model.ScriptLibraryPUTRequest;
+import org.wso2.carbon.identity.api.server.script.library.v1.model.ScriptLibraryResponse;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.functions.library.mgt.exception.FunctionLibraryManagementClientException;
+import org.wso2.carbon.identity.functions.library.mgt.exception.FunctionLibraryManagementException;
+import org.wso2.carbon.identity.functions.library.mgt.exception.FunctionLibraryManagementServerException;
+import org.wso2.carbon.identity.functions.library.mgt.model.FunctionLibrary;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.api.server.common.Constants.V1_API_PATH_COMPONENT;
+import static org.wso2.carbon.identity.api.server.script.library.common.Constants.SCRIPT_LIBRARY_CONTENT_PATH;
+import static org.wso2.carbon.identity.api.server.script.library.common.Constants.SCRIPT_LIBRARY_PATH_COMPONENT;
+
+/**
+ * Call internal osgi services to perform server script library management operations.
+ */
+public class ServerScriptLibrariesService {
+
+    private static final Log log = LogFactory.getLog(ServerScriptLibrariesService.class);
+    private static final int DEFAULT_OFFSET = 0;
+
+    /**
+     * Get list of Script Libraries.
+     *
+     * @param limit  Items per page.
+     * @param offset Offset.
+     * @return ScriptLibraryListResponse.
+     */
+    public ScriptLibraryListResponse getScriptLibraries(Integer limit, Integer offset) {
+
+        List<FunctionLibrary> functionLibraries;
+        try {
+            functionLibraries = ScriptLibraryServiceHolder.getScriptLibraryManagementService()
+                    .listFunctionLibraries(ContextLoader.getTenantDomainFromContext());
+        } catch (FunctionLibraryManagementException e) {
+            throw handleScriptLibraryError(e, Constants.ErrorMessage.ERROR_CODE_ERROR_LISTING_SCRIPT_LIBRARIES);
+        }
+        return createScriptLibrariesList(functionLibraries, validateLimit(limit), validateOffset(offset));
+    }
+
+    /**
+     * Validate the offset.
+     *
+     * @param offset Offset value
+     * @return Validated offset.
+     */
+    private int validateOffset(Integer offset) {
+
+        if (offset != null && offset >= 0) {
+            return offset;
+        } else {
+            return DEFAULT_OFFSET;
+        }
+    }
+
+    /**
+     * Validate the limit.
+     *
+     * @param limit Limit value.
+     * @return Validated limit.
+     */
+    private int validateLimit(Integer limit) {
+
+        final int maximumItemPerPage = IdentityUtil.getMaximumItemPerPage();
+        if (limit != null && limit > 0 && limit <= maximumItemPerPage) {
```

Comment:
let's throw an error (Bad request ) if limit is negative. Return default value for null limit

Reply:
[Fixed commit](https://github.com/wso2/identity-api-server/pull/81/commits/f1f24a495063022b48d0fe2a91be8b83c6cf07de)

Reply:
this should be modified as 

if limit is null --> return default (IdentityUtil.getDefaultItemPerPage())
else if limit negative --> throw error
else if limit is greater than maximumItemPerPage, then return maximumItemPerPage
else return the limit

Reply:
Negative values are checked from an upper level.
![](https://i.imgur.com/o1izMOg.png)

Reply:
Fixed other validations